### PR TITLE
Implement a notebook with a working version of PPO

### PR DESCRIPTION
**Pull request by gth828r**
* Ensure that we run on GPUs
 * Implement an example actor critic model to use for testing
 * Implement TDD-lambda return estimator for computing the general
   advantage estimate
 * Implement the PPO loss function
 * Implement a minibatch training loop which runs in graph execution
   mode
 * Create a test case of gathering data and training over CartPole-v1
 * Instantiate an actor critic model to use
   * Adam optimizer chosen because it allegedly works faster, according
     to the original paper and the tasks learned there
   * MSE error used for the critic loss, because the paper suggests it,
     and Huber loss was resulting in divergence during training
   * Two hidden layers used, although no testing was performed to see if
     this was necessary
 * Create an inital set of testing hyper-parameters which seem to work
   consistently on GPUs and CPUs

Addresses #1

### CODE REVIEW

**Comment by gth828r** [comment]
I sanity checked the notebook with what was last run, and it looks good. For full disclosure, I should mention that the rendered policy after training is by far the best policy I saw. But in general, I think that the latest implementation was consistently meeting the early stopping criteria for CartPole.

More formal conversion of this code into a python module is to be done in #7, and this implementation of PPO appears to work, so I think this one should be good to go!
